core/foundation/src/TClassEdit.cxx

```diff
@@ -838,16 +838,23 @@ void TClassEdit::GetNormalizedName(std::string &norm_name, std::string_view name
    }
 
    norm_name = std::string(name); // NOTE: Is that the shortest version?
+
+   if (TClassEdit::IsArtificial(name)) {
+      // If there is a @ symbol (followed by a version number) then this is a synthetic class name created
+      // from an already normalized name for the purpose of supporting schema evolution.
+      return;
+   }
+
    // Remove the std:: and default template argument and insert the Long64_t and change basic_string to string.
    TClassEdit::TSplitType splitname(norm_name.c_str(),(TClassEdit::EModType)(TClassEdit::kLong64 | TClassEdit::kDropStd | TClassEdit::kDropStlDefault | TClassEdit::kKeepOuterConst));
    splitname.ShortType(norm_name, TClassEdit::kDropStd | TClassEdit::kDropStlDefault | TClassEdit::kResolveTypedef | TClassEdit::kKeepOuterConst);
 
-   if (splitname.fElements.size() == 3 && (splitname.fElements[0] == "std::pair" || splitname.fElements[0] == "pair")) {
```

Comment:
In TTreeGeneratorBase.cxx, I left 3 or 4 because the likely wrong one was there with no (obvious negative consequence) and (I should add a comment) thus it might actually be the right one .... i.e. to remove it I would have to try to produce a test and the code is used little enough to not be a priority (to spend the time to write the test).

Reply:
A comment would be great! :-)

Reply:
> without any comment. 

Well that's part of the document contract of TClassEdit::TSplitType :)  

But nevertheless I will add a comment.